src/util/scrollbar.ts

```diff
@@ -0,0 +1,44 @@
+import {Injectable, Inject} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+
+
+
+@Injectable()
+export class ScrollBar {
+  constructor(@Inject(DOCUMENT) private _document) {}
+
+  compensate() {
+    if (!this._isPresent()) {
+      return () => {};
+    }
+    const width = this._getWidth();
+    return this._adjustBody(width);
+  }
+
+  private _adjustBody(width: number) {
+    const {body} = this._document;
+    const userSetPadding = body.style.paddingRight;
+    const paddingAmount = parseFloat(window.getComputedStyle(body)['padding-right']);
+    body.style['padding-right'] = `${paddingAmount + width}px`;
+    return () => body.style['padding-right'] = userSetPadding;
+  }
+
+  private _isPresent(): boolean {
+    const rect = this._document.body.getBoundingClientRect();
+    return rect.left + rect.right < window.innerWidth;
+  }
+
+  private _getWidth(): number {
```

Comment:
This could be called only once, if I'm not mistaken? (I mean, we don't have to call it in each and every `compensate()` call.

Reply:
I don't know... could there be any situation where the width of the scrollbar would change (with custom styling then) throughout the application lifetime? (and would we support that?)

If not, I can cache the result.

Reply:
Let's leave it as-is for now.